mods/tnt/init.lua

```diff
@@ -415,6 +416,9 @@ minetest.register_node("tnt:gunpowder", {
 	on_blast = function(pos, intensity)
 		tnt.burn(pos, "tnt:gunpowder")
 	end,
+	on_burn = function(pos)
+		minetest.set_node(pos, {name = "tnt:gunpowder_burning"})
```

Comment:
I guess so that any gunpowder on the edge of an explosion is lit. However i'm ok with gunpowder just being removed 'on blast'.